tests/unit/test_job_submitter.py

```diff
@@ -19,162 +23,258 @@
-    args, setup = _create_setup(tmp_path, mask_content=mask, mask_file=False)
-    setup.update(run_local=True)
-    job_submit(**setup)
-    _test_output(args)
+    setup = InputParameters(working_directory=tmp_path, run_local=True)
+    setup.create_mask(content=mask, as_file=False)
+    job_submit(**asdict(setup))
+    _test_output(setup)
 
 
 @run_only_on_linux
 @pytest.mark.parametrize("maskfile", [True, False])
 def test_job_creation_and_dryrun(tmp_path, maskfile):
-    args, setup = _create_setup(tmp_path, mask_file=maskfile)
-    setup.update(dryrun=True)
-    job_submit(**setup)
+    """ Tests that the jobs are created as dry-run from mask-file and from mask-string. """
+    setup = InputParameters(working_directory=tmp_path, dryrun=True)
+    setup.create_mask(as_file=maskfile)
+    job_submit(**asdict(setup))
     _test_subfile_content(setup)
-    _test_output(args, post_run=False)
+    _test_output(setup, post_run=False)
 
 
 @run_only_on_linux
 @pytest.mark.parametrize("maskfile", [True, False])
 def test_find_errorneous_percentage_signs(tmp_path, maskfile):
+    """ Tests that a key-error is raised on a mask-string with percentage signs, 
+    that are not part of the replacement parameters. """
     mask = "%(PARAM1)s.%(PARAM2)d\nsome stuff # should be 5%\nsome % more % stuff."
-    args, setup = _create_setup(tmp_path, mask_content=mask, mask_file=maskfile)
+    setup = InputParameters(working_directory=tmp_path)
+    setup.create_mask(content=mask, as_file=maskfile)
     with pytest.raises(KeyError) as e:
-        job_submit(**setup)
-    assert "problematic '%'" in e.value.args[0]
+        job_submit(**asdict(setup))
+    assert "problematic '%'" in str(e)
 
 
 @run_only_on_linux
 @pytest.mark.parametrize("maskfile", [True, False])
 def test_missing_keys(tmp_path, maskfile):
+    """ Tests that a key-error is raised on a mask-string with missing keys in the replacement dict. """
     mask = "%(PARAM1)s.%(PARAM2)s.%(PARAM3)s"
-    args, setup = _create_setup(tmp_path, mask_content=mask, mask_file=maskfile)
+    setup = InputParameters(working_directory=tmp_path)
+    setup.create_mask(content=mask, as_file=maskfile)
     with pytest.raises(KeyError) as e:
-        job_submit(**setup)
-    assert "PARAM3" in e.value.args[0]
+        job_submit(**asdict(setup))
+    assert "PARAM3" in str(e)
 
 
 @run_if_not_linux
 def test_not_on_linux(tmp_path):
-    args, setup = _create_setup(tmp_path)
+    """ Test that an error is raised if htcondor bindings are not found.
+    If this tests fails, this might mean, that htcondor bindings are finally 
+    available for the other platforms. """
+    setup = InputParameters(working_directory=tmp_path)
+    setup.create_mask()
     with pytest.raises(EnvironmentError) as e:
-        job_submit(**setup)
-    assert "htcondor bindings" in e.value.args[0]
+        job_submit(**asdict(setup))
+    assert "htcondor bindings" in str(e)

```

Comment:
I would take the opportunity to rename this guy too, maybe `test_nonlinux_run_raises` or something